vm/src/stdlib/socket.rs

```diff
@@ -170,7 +168,7 @@ impl PySocket {
         PyRwLockReadGuard::try_map(self.sock.read(), |sock| sock.get()).ok()
     }
 
-    fn sock_io(&self) -> io::Result<PyMappedRwLockReadGuard<'_, Socket>> {
+    pub(crate) fn sock_io(&self) -> io::Result<PyMappedRwLockReadGuard<'_, Socket>> {
```

Comment:
Maybe it would make sense to expose a pub `send(&[u8]) -> io::Result<usize>` and `recv(&mut [u8]) -> io::Result<usize>` ~~for once the stdlib gets moved out of `rustpython-vm`~~ oh I guess `socket` is also in stdlib. But the above still seems like a better abstraction than exposing `ReadGuard`.

Reply:
(and flush() for sockets is a no-op)

Reply:
thanks. I made pub read, write and flush in PySocket